Pure vanilla app

nweldev opened the pull request:
## Objectif

App 100% vanilla avec Webpack, sans autre dépendance que MDL. En testant plus avec lit-html, je me suis rendu assez rapidement compte que la librairie rendait les choses assez restrictives pour qu'on se retrouve quasiment avec un projet Polymer si on voulait bien faire ... et que je m'éloignait donc de plus en plus du vanilla.

Voir #17 pour les discussions globales sur la mono-branche.

## Inspirations et démarche

Très librement inspiré de https://github.com/GoogleChromeLabs/webpack-training-project, afin de s'approcher au mieux des différents labs performance que nous voudrons adapter, sans utiliser preact pour autant (non pas que ça n'aurait pas été un bon choix, mais nous nous sommes fixés sur du Vanilla). Ce projet ne gérait cependant pas le re-rendering et le state routing donc j'ai du repenser pas mal la chose. J'ai préféré également m'orienter sur de la POO façon Angular plutôt qu'une approche fonctionnelle façon React pour plus de lisibilité même si cela n'est pas idéal niveau perf et maintenabilité, vu qu'on est dans tous les cas sur des imbrications de fonctions avec une responsabilité de rendering ascendante.

Pour l'instant conservé comme un step à part histoire de pouvoir continuer à lancer l'app angularjs pour comparer. Utiliser `npm run dev:server` pour tester.

Bien entendu, rien n'est vraiment optimisé, afin de permettre de faire cela dans les exercices.

## TO DO
 
* [x] Autres components et routes
* [x] ajouter un routage paramétré dans utils/Router
* [x] refact la gestion des peoples en ajoutant le filtering
* [x] gestion des data-link dans l'app comme pour la landing page
* [x] revoir la structure du repo
* [x] mettre à jour les fichiers de chaque step
* [ ] App-shell (steps sw 5-1 et 5-2) : mettre à jour le support pour un chapitre plus complet OU mettre à jour le projet pour faire un "mauvais" app-shell (avec composants en sans bundling) pour permettre d'avoir une liste de fichiers nécessaires ? voir avec @dahfazz en fonction de ce qu'il préfère faire pour la partie material design
* [ ] documenter (oui parce que évidemment j'ai rien documenté comme un gros sale mais ça viendra :wink:)
* [x] transformer les templates angular-material vers mdl
  * tâche la plus lourde à priori, mais nous avons ce soucis avec n'importe quelle autre solution
  * je pense que ce pourrait être l'occasion dés maintenant tant qu'à faire, vu que je n'ai pas fait grand chose avec MDL pour l'instant, de passer plutôt à https://github.com/material-components/material-components-web et en sass. ***Qu'en pensez vous ?***
  * **Help Wanted: Si il y a des volontaires, allez-y,  car après quelques essais j'ai vite saturé :sweat_smile:** @dahfazz je sais que c'est ton kiff :stuck_out_tongue: 
* [x] améliorer la gestion et config webpack pour les steps
* [x] supprimer le projet angularjs
* [x] Migration des exercices déjà existant à la sauce vanilla et compatible webpack
* [x] Device API 
  * [x] Geolocation API
  * [x] Fullscreen -> **Démo à Faire ?** 
* [-] Finir les slides
  * [ ] Event dans Push
  * [ ] Web Authentification
  * [ ] Gif is dead

Le reste sera pour les steps à priori, surtout sur la perf.

nlm-pro committed:
add a new vanilla app base

nlm-pro committed:
reproduce home component in vanilla

nlm-pro committed:
pleopleCard conversion to vanilla without md

nlm-pro committed:
split PeopleCart with pseudo-components for more readability

noelmace commented:
Voir #17 pour les discussions globales sur la mono-branche.

noelmace commented:
Ah oui et évidemment j'ai rien documenté comme un gros sale mais ça viendra :wink:

nlm-pro committed:
add router params

nlm-pro committed:
add people route and fix some Router issues

nlm-pro committed:
add people search feature in vanilla app

nweldev commented:
Pas de véritable refactoring sur people pour l'instant, car je suis un peu pressé par le temps, mais le filtering est là. Je crois que du coup toutes les features de l'app d'origine angularjs sont couvertes.

Je laisse donc le plaisir de reprendre les templates html avec (ou sans, c'est encore mieux) material components web et le css (ou scss, comme il lui semblera le plus pertinent)  à @dahfazz et m'attaquerai à la finalisation du reste dans la foulée.

nlm-pro committed:
move common/js to common/angularjs

nlm-pro committed:
move assets to theire how folder

nlm-pro committed:
move step-vanilla to common/app

manekinekko commented:
Je suis un peu perplexe car je sais que ce changement part d'une bonne intention, mais j'ai quelques reserves...

Je trouve que le fait de vouloir partir sur une app vanilla js est une très bonne idée. Mais le fait de réimplémenter les mêmes fonctionnalités qu'offrent les autres frameworks (angular) et lib (react) et en introduisant les mêmes outils de build (webpack) ne résout en rien le problème initiale. Au contraire, cela nous éloigne beaucoup du sujet de la sfeir school, qu'est : les PWA (hors frameworks) !

Je comprends également que le but est d'avoir une app de démo pour illustrer les différents concepts des sfeir school. Or, d'après les retours que j'ai eu sur d'autres sfeir school, type "firebase";  L'application sfeir people n'est pas adaptée lorsqu'on n'est en dehors des sfeir school "frameworks". La compléxité ne provient pas forcément des frameworks (pas que) mais elle vient surtout de l'application sfeir people.

Je suis donc d'avis qu'on parte plutôt sur une nouvelle application avec beaucoup moins de fonctionnalités et qui nous permetterait de mettre le focus sur les concepts enseignés. Pourquoi à votre avis la TODO app est devenue si célébre ;)

manekinekko reviewed (dismissed):
See: https://github.com/Sfeir/pwa-200/pull/18#issuecomment-416566463

nlm-pro committed:
update step-1 and webpack config

nweldev commented:
Hello @manekinekko.

Oui en effet, tu n'as pas été dans la boucle de préparation de ce nouveau format.
Big :+1: sur le fait de partir sur une nouvelle application en ce qui me concerne, mais nos échanges avec @cbalit et des contraintes de délais nous amènent, dans un premier temps, à privilégier une réutilisation maximale de l'existant en ce qui concerne l'app de base. Le passage en vanilla et avec webpack a pour but d'éliminer des pré-requis et de faciliter le passage de la formation en 2 jours avec notamment un plus fort focus sur les perfs, tout en se rapprochant des exemples Google, et des cas réels que rencontrent les étudiants (d'où webpack pour ces 3 raisons).

En ce qui concerne le dev en lui même, quel est ton feedback exactement ? Il ne me semble pas de mon point de vue avoir ré-implémenté des fonctionnalités de frameworks type Angular ou lib type React, mais je suis impatient d'avoir des retours sur ces points.

Sinon concernant les sujets de la School en elle même, je t'invite à consulter le nouveau support sur le drive (en cours de rédaction, donc regarde surtout le plan) : on abordera bien les frameworks et libs. 

L'objectif étant (on a pas rédigé donc ça n'engage que moi et j'invite les autres à me corriger si besoin) : 
1. de mettre l'accent sur le fait que Progressive est avant tout une approche et un ensemble de bonnes pratiques, et donc ni une solution technique ni une cible
2. en suivant cette optique, mettre l'accent sur les compétences fondamentales pour aborder les PWA (connaissance du browser et de son public) et être plus exhaustif tout en ce démarquant des autres formations sur le marché
3. donner l'envie aux élèves d'appliquer cela à leur retour, et donc être contraint par un minimum de technos dans un premier temps puis, dans un second temps, leur donner les armes pour appliquer ça dans leur cadre à eux (le plus souvent, framework) ... d'où l'envie, en effet, de passer à une autre app asap :smiley:

nlm-pro committed:
vanilla step 1 solution

nlm-pro committed:
rename steps in order to be more consistent with slides

nlm-pro committed:
vanilla sw step 2

nlm-pro committed:
vanilla sw step 3

nlm-pro committed:
vanilla sw step 4

nlm-pro committed:
vanilla: default files by steps types

nlm-pro committed:
vanilla: steps sw 5-1 and 2

Update the landing (index) template in order to have better files to cache for the app shell.
Also fix some issues with previous steps.

nlm-pro committed:
vanilla: steps 5-3 to 5

jefBinomed committed:
vanilla: steps 6 to manifest

jefBinomed committed:
vanilla: steps 7 to install

jefBinomed committed:
vanilla : steps 8 to push

jefBinomed committed:
vanilla : steps 9 to workbox

jefBinomed committed:
vanilla : Fix basic steps run

jefBinomed committed:
angular : fix rename of steps

jefBinomed committed:
vanilla : fix style problems

jefBinomed committed:
vanilla: fix styles problems

jefBinomed committed:
vanilla: migrate angular to vanilla for step manifest

jefBinomed committed:
vanilla: migrate angular to vanilla for step install

jefBinomed committed:
vanilla: fix service worker urls for steps install & manifest

jefBinomed committed:
vanilla : remove angular for step push

jefBinomed committed:
vanilla: migrate to vanilla solution for step workbox

jefBinomed committed:
Vanilla : fix some bugs due to migration to vanilla

jefBinomed committed:
vanilla: change `npm run dev:server`  to `npm start` because gulp is not needed anymore

jefBinomed committed:
vanilla: Fix minor css pb with search field

jefBinomed committed:
vanilla: Add Coordinates of cities for Geoloc exercise

jefBinomed committed:
vanilla: refactor router to have context router (pb with webpack else)

jefBinomed committed:
vanilla: modification to webpack to point to the good entry point

jefBinomed committed:
vanilla: add device-1 exercice (geolocation)

jefBinomed committed:
vanilla : clean Angular code

jefBinomed committed:
vanilla: Add fullscreen exercice

nlm-pro committed:
add contributor Noël Macé

nlm-pro committed:
fix: upgrade dom for materialjs after People rendering

nlm-pro committed:
fix code typo in index html

nlm-pro committed:
delete old files and dependencies

nlm-pro committed:
fix dependencies versions

jefBinomed committed:
vanilla : add missing css to service worker

jefBinomed committed:
Merge remote-tracking branch 'origin/pure-vanilla' into pure-vanilla

nlm-pro committed:
add dev script for windows

nlm-pro committed:
delete old index.html files

nlm-pro committed:
simplify sw4 solution
